core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1081,7 +1080,38 @@ else if (!current.isEnabled()) {
 
             return deps;
         }
-        
+
+
+        /**
+         * Returns a list of plugins which need to be installed or upgraded for this plugin to work (include transient dependencies too).
+         * Prevent issue JENKINS-36239, since it contains all dependencies and it is not necessary to call it recursively
+         */
+        public Set<Plugin> getAllDependencies(){
+            //map of dependencies for plugin and its dependencies
+            Map<Plugin,List<Plugin>> allDependencies = new HashMap<Plugin,List<Plugin>>();
+            return addMissingDependencies(allDependencies);
+        }
+
+        private Set<Plugin> addMissingDependencies(Map<Plugin, List<Plugin>> plugins) {
+            if (!plugins.containsKey(this)) {
+                plugins.put(this, getNeededDependencies());
```

Comment:
Hello, good point I will write test for it and do necessary changes. Thank you for review, it is really welcome since this is jenkins key feature - better look ten times then break deployment of plugins.